tests/unit/components/status/HeatMapColorBar.spec.js

```diff
@@ -0,0 +1,278 @@
+    );
+  });
+  test("Given that the uuid is empty, upper_range, lower_range, height, and gradient_range, When mounted successfully, Then assert that upper_range,lower_range, height are applied on HeatMapColorBar and as uuid is empty then gradient_range values are applied on linear-gradient", async () => {
+    const propsData = {
+      upper_range: "100",
+      lower_range: "0",
+      gradient_uuid: "",
+      heatmap_height: 481,
+      gradient_range: [
+        { color: "#2c7bb6", offset: "0%" },
+        { color: "#00a6ca", offset: "25%" },
+        { color: "#00ccbc", offset: "50%" },
+        { color: "#90eb9d", offset: "75%" },
+        { color: "#ffff8c", offset: "100%" },
+      ],
+    };
+
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      localVue,
+    });
+    const target_span_higher = wrapper.find(
+      ".span__heatmap-scale-higher-value"
+    );
+    expect(target_span_higher.text()).toStrictEqual("100 μN");
+    const target_span_lower = wrapper.find(".span__heatmap-scale-lower-value");
+    expect(target_span_lower.text()).toStrictEqual("0 μN");
+
+    const target_div_gradient_holder = wrapper.find(
+      ".div__heatmap-gradient-holder"
+    );
+    expect(target_div_gradient_holder.attributes().style).toBe(
+      "height: 481px;"
+    );
+
+    const renderwrapper = await renderToString(ComponentToTest, {
+      propsData,
+      localVue,
+    });
+    expect(renderwrapper).toContain(
+      '<div data-server-rendered="true"><div class="div__heatmap-gradient-holder" style="height:481px;background:linear-gradient(to top, #2c7bb6 0%,#00a6ca 25%,#00ccbc 50%,#90eb9d 75%,#ffff8c 100%);"></div> <span class="span__heatmap-scale-higher-value">100 μN</span> <span class="span__heatmap-scale-lower-value" style="top:465px;">0 μN</span></div>'
+    );
+  });
+
+  test("Given that the uuid, upper_range, lower_range and height, When mounted successfully, Then prop update of the heatmap_height to 400, assert height is updated to 400 and as uuid is match a default linear-gradient is applied on HeatMapColorBar", async () => {
+    const propsData = {
+      upper_range: "100",
+      lower_range: "0",
+      gradient_uuid: "0f81155d-23ec-4790-a3fe-92a6cc7c3c47",
+      heatmap_height: 481,
+    };
+    wrapper = await mount(ComponentToTest, {
+      propsData,
+      localVue,
+    });
+
+    const target_div_gradient_holder = wrapper.find(
+      ".div__heatmap-gradient-holder"
+    );
+
+    expect(target_div_gradient_holder.attributes().style).toBe(
+      "height: 481px;"
+    );
+
+    await wrapper.setProps({ heatmap_height: 400 });
+
+    await wrapper.vm.$nextTick();
+
+    expect(target_div_gradient_holder.attributes().style).toBe(
+      "height: 400px;"
+    );
+
+    propsData.heatmap_height = 400;
```

Comment:
is there a reason you are setting the value here?  isn't this what `await wrapper.setProps({ heatmap_height: 400 });` is supposed to accomplish?

This comment applies to other tests as well

Reply:
As the JEST and Vue-test-utils doesn't provide the essential asserting and reactive feature of the component. The question is for web artifacts, there are no testing required as per the general opinion and its a redundant testcase